src/prefect/orion/services/loop_service.py

```diff
@@ -103,39 +107,47 @@ async def start(self, loops=None) -> None:
             )
             self.logger.debug(f"Finished running {self.name}. Next run at {next_run}")
 
-            # check the `should_stop` flag every 1 seconds until the next run time is reached
-            while pendulum.now("UTC") < next_run and not self.should_stop:
+            # check the `_should_stop` flag every 1 seconds until the next run time is reached
+            while pendulum.now("UTC") < next_run and not self._should_stop:
                 await asyncio.sleep(
                     min(1, (next_run - pendulum.now("UTC")).total_seconds())
                 )
 
-        await self.shutdown()
+        await self._on_stop()
 
-    async def stop(self) -> None:
+    async def stop(self, block=True) -> None:
```

Comment:
IMO the default should make this safe to call without understanding, but I'm not as familiar with all the places this code is used.

Reply:
I think the _typical_ use for this (do we even know that?) is to `await service.stop()` from _outside_ I think.

Reply:
I made this change -- I think in looking at the codebase, services are almost always run as long-running services and Prefect itself is really the only actor that needs to start / stop them. However, developers of services may want to call `.stop()` to create an early stopping behavior (as @cicdw did)